internal/winapi/winapi.go

```diff
@@ -0,0 +1,61 @@
+package winapi
+
+import "github.com/Microsoft/go-winio/pkg/guid"
+
+//go:generate go run ..\..\mksyscall_windows.go -output zsyscall_windows.go winapi.go
+
+//sys LocalAlloc(flags uint32, size int) (ptr uintptr) = kernel32.LocalAlloc
+//sys LocalFree(ptr uintptr) = kernel32.LocalFree
+
+//sys NTCreateFile(handle *uintptr, accessMask uint32, oa *ObjectAttributes, iosb *IOStatusBlock, allocationSize *uint64, fileAttributes uint32, shareAccess uint32, createDisposition uint32, createOptions uint32, eaBuffer *byte, eaLength uint32) (status uint32) = ntdll.NtCreateFile
+//sys NTSetInformationFile(handle uintptr, iosb *IOStatusBlock, information uintptr, length uint32, class uint32) (status uint32) = ntdll.NtSetInformationFile
+
+//sys NTOpenDirectoryObject(handle *uintptr, accessMask uint32, oa *ObjectAttributes) (status uint32) = ntdll.NtOpenDirectoryObject
+//sys NTQueryDirectoryObject(handle uintptr, buffer *byte, length uint32, singleEntry bool, restartScan bool, context *uint32, returnLength *uint32)(status uint32) = ntdll.NtQueryDirectoryObject
+
+//sys RtlNtStatusToDosError(status uint32) (winerr error) = ntdll.RtlNtStatusToDosError
+
+//sys CMGetDeviceIDListSize(pulLen *uint32, pszFilter *byte, uFlags uint32) (hr error) = cfgmgr32.CM_Get_Device_ID_List_SizeA
+//sys CMGetDeviceIDList(pszFilter *byte, buffer *byte, bufferLen uint32, uFlags uint32) (hr error)= cfgmgr32.CM_Get_Device_ID_ListA
+//sys CMLocateDevNode(pdnDevInst *uint32, pDeviceID string, uFlags uint32) (hr error) = cfgmgr32.CM_Locate_DevNodeW
+//sys CMGetDevNodeProperty(dnDevInst uint32, propertyKey *DevPropKey, propertyType *uint32, propertyBuffer *uint16, propertyBufferSize *uint32, uFlags uint32) (hr error) = cfgmgr32.CM_Get_DevNode_PropertyW
+
+type IOStatusBlock struct {
+	Status, Information uintptr
+}
+
+type ObjectAttributes struct {
+	Length             uintptr
+	RootDirectory      uintptr
+	ObjectName         uintptr
+	Attributes         uintptr
+	SecurityDescriptor uintptr
+	SecurityQoS        uintptr
+}
+
+type UnicodeString struct {
+	Length        uint16
+	MaximumLength uint16
+	Buffer        uintptr
+}
+
+type ObjectDirectoryInformation struct {
+	Name     UnicodeString
+	TypeName UnicodeString
+}
+
+type FileLinkInformation struct {
+	ReplaceIfExists bool
+	RootDirectory   uintptr
+	FileNameLength  uint32
+	FileName        [1]uint16
+}
+
+type FileDispositionInformationEx struct {
+	Flags uintptr
+}
+
+type DevPropKey struct {
+	Fmtid guid.GUID
+	Pid   uint32
+}
```

Comment:
Can we move all of this into a separate file in the winapi package and same for the rest? Having all of the definitions and structs from different dlls in one file feels odd. For example can we move this to a file named cfgmgr32.go (or something else that's easier to identify)? Then just edit the go:generate command to 
```go
//go:generate go run ..\..\mksyscall_windows.go -output zsyscall_windows.go winapi.go cfgmgr32.go ntdll.go kernel32.go
```

Reply:
I think the point of Kevin's suggestion (among other things) was to move away from caring about implementation details of given windows api calls, such as which dll they happen to be in. With that in mind, I don't think it makes sense to separate the files out here by dll.

Reply:
I suppose, just feels weird to have a mish mash of n different dlls/struct definitions in one file. I don't see how separating makes this any more of a hassle but I really don't care either way, I just like how it's structured in go-winio 😃

Reply:
@kevpar thoughts here?

Reply:
My main point was the exported interface shouldn't expose the details of how (or where) things are implemented, but instead focus on the functionality that it provides.

Within the `winapi` package itself, we should organize the code however makes the most sense for readability and comprehension. I think it would be fine to separate out individual files in the package based on what DLL they are working with. As this package grows, we will probably want to have things separated out more than than just between `winapi.go` and `utilities.go`, but I don't have strong opinions on if sorting by DLL or some other rubric makes the most sense.

Reply:
I've split the calls for each dll into a separate file. Let me know if there are any thoughts on this.

Reply:
Looks good :) DevPropKey is cfgmgr related correct? I'd still move the related consts and structs to their relevant files if they only apply to that api.